### PR TITLE
Ccgo WIN

### PR DESCRIPTION
**Pull request by steffengy**
This adds win support to the CRT.

There are ~3 tests that do not pass on AMD64 (int -> int64 coercion, minor stuff).
SQLite Tests pass.

I have only run the SQLite tests for i386, since the cross-compiled tests would take
several hours to run (magnitutde slower than x64 tests on x86_64)

### CODE REVIEW

File: windows.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
+// Copyright 2017 The Virtual Authors. All rights reserved.
```
Comment (cznic) [inline windows.go]: s/Virtual/CRT/

File: windows_386.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
+// Copyright 2017 The Virtual Authors. All rights reserved.
```
Comment (cznic) [inline windows_386.go]: s/Virtual/CRT/

File: windows_386.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:generate go run generator.go
```
Comment (cznic) [inline windows_386.go]: I think this line has to be removed.

File: windows_amd64.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
+// Copyright 2017 The Virtual Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:generate go run generator.go
```
Comment (cznic) [inline windows_amd64.go]: I think this line has to be removed.

File: windows_amd64.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
+// Copyright 2017 The Virtual Authors. All rights reserved.
```
Comment (cznic) [inline windows_amd64.go]: s/Virtual/CRT/

File: windows_386.go (hunk 3 above)
Comment (steffengy) [reply windows_386.go]: You're right. should only be in windows.go

File: windows_impl_386.go [hunk 6]
```diff
@@ -0,0 +1,1343 @@
...
+
+import (
+	"fmt"
+	"unsafe"
+	"syscall"
+	"os"
+)
+
+var (
+	modkernel32           = syscall.NewLazyDLL("kernel32.dll")
```
Comment (cznic) [inline windows_impl_386.go]: Please gofmt.

File: windows_impl_amd64.go [hunk 7]
```diff
@@ -0,0 +1,1343 @@
...
+
+import (
+	"fmt"
+	"unsafe"
+	"syscall"
+	"os"
+)
+
+var (
+	modkernel32           = syscall.NewLazyDLL("kernel32.dll")
```
Comment (cznic) [inline windows_impl_amd64.go]: Please gofmt.

**Review by cznic — CHANGES_REQUESTED** [review]
A couple of small changes please.

**Comment by steffengy** [comment]
Done

**Review by cznic — APPROVED** [review]
LGTM, thanks.

**Pull request merged**